### PR TITLE
HZN-375: Added option to specify nameserver to DnsResolutionMonitor

### PR DESCRIPTION
**Pull request by fooker**
Issue: http://issues.opennms.org/browse/HZN-375.
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS96

This is the same as #329 but against develop.

### CODE REVIEW

**Comment by soleger** [comment]
I would prefer that the reason codes were changed like I mentioned in the last PR. The case where no result is returned is covered by the "Unable to resolve..." message in the old code. Am I reading this incorrectly?

**Comment by fooker** [comment]
I'Ve changed the reason codes again. Do you think it's better, now?

File: opennms-services/src/main/java/org/opennms/netmgt/poller/monitors/DNSResolutionMonitor.java [hunk 1]
```diff
@@ -28,115 +28,113 @@
...
+
+        // Resolving succeeded - checking results
+        final long end = System.currentTimeMillis();
+
+        // Check if result is valid
+        if (!ipv4Found && !ipv6Found) {
+            return PollStatus.unavailable("Unable to resolve host '" + name + "'");
+
+        } else  if (ipv4Required && !ipv4Found) {
+            return PollStatus.unavailable("No IPv4 Address result (A record) returned for '" + name + "'");
```
Comment (soleger) [inline opennms-services/src/main/java/org/opennms/netmgt/poller/monitors/DNSResolutionMonitor.java]: In the old monitor, this was:
```java
name + " could only be resolved to an IPv6 address"
```
indicating that an AAAA record was found but no A record was found. This is not apparent in the new message. Can we make the messages for this something like:
```java
name + " could be resolved to an IPv6 address (AAAA record) but not an IPv4 address (A record)."
```

**Comment by fooker** [comment]
Ok, now I got what you are talking about - sorry. Does this look better, now?

**Pull request merged**